depends/packages/boost.mk

```diff
@@ -1,10 +1,29 @@
 package=boost
-$(package)_version=1.81.0
-$(package)_download_path=https://archives.boost.io/release/$($(package)_version)/source/
-$(package)_file_name=boost_$(subst .,_,$($(package)_version)).tar.gz
-$(package)_sha256_hash=205666dea9f6a7cfed87c7a6dfbeb52a2c1b9de55712c9c1a87735d7181452b6
+$(package)_version=1.85.0
+$(package)_download_path=https://github.com/boostorg/boost/releases/download/boost-$($(package)_version)
+$(package)_file_name=boost-$($(package)_version)-cmake.tar.gz
+$(package)_sha256_hash=ab9c9c4797384b0949dd676cf86b4f99553f8c148d767485aaac412af25183e6
+$(package)_build_subdir=build
+
+# This compiles a few libs unnecessarily because date_time and test don't have
```

Comment:
Are we going to upstream this? Suprised there's not already an option to do this for at least `date_time`, given it's a stub library that only exists for back compat. Wonder if we can (easily) patch it out in the interim.

Reply:
Blah. I started working on it, but there's no precedent for a `cmake -DHEADER_ONLY` in the boost libs and their CMake files are auto-generated. So I assumed there'd be resistance to a patch like this.

But now that we have a reason, sure, I'll patch them and PR the patches and see how it goes.

Reply:
> So I assumed there'd be resistance to a patch like this.

Given some recent happenings, that may no-longer be the case. Seems like there's quite a substantial effort to newly "modularize" Boost. See mailing list discussion here: https://lists.boost.org/Archives/boost/2024/01/255704.php and [WIP] overview here: https://github.com/grafikrobot/boost-b2-modular/blob/b2-modular/README.adoc.